File: bindings/python/examples/provider_query.py
from __future__ import annotations

from _common import (
    load_openai_provider_client,
    print_text,
    provider_endpoint,
    read_provider_args,
    text_options,
)


def main() -> None:
    model, prompt = read_provider_args("Write one sentence about provider inference.")
    client = load_openai_provider_client(model)
    run = client.query(
        prompt,
        endpoint=provider_endpoint(model),
        options=text_options(),
    )
    print_text(run.result())


if __name__ == "__main__":
    main()

File: bindings/node/examples/provider_query.mjs
import {
  loadOpenAiProviderClient,
  printTextRun,
  providerEndpoint,
  readProviderArgs,
  textOptions,
} from './_common.mjs';

const { model, input } = readProviderArgs('Write one sentence about provider inference.');
const client = loadOpenAiProviderClient(model);
await printTextRun(client.query({
  endpoint: providerEndpoint(model),
  prompt: input,
  options: textOptions(),
}));
